server/src/main/scala/za/co/absa/atum/server/api/database/Runs.scala

```diff
@@ -35,33 +37,34 @@ class Runs (implicit dBEngine: SlickPgEngine) extends DBSchema{
 
 object Runs {
 
-  private def scalaIterableToSQLArray(toConvert: Seq[Any]): String = {
-    SerializationUtils.asJson(toConvert)
-      .replace("[", "{")
-      .replace("]", "}")
+  private def nestedScalaSeqToPgArray[T <: AnyRef: ClassTag](toConvert: Seq[Seq[T]]): String = {
+    val scalaSeqJsonized = toConvert.map(scalaSeqToPgArray).mkString(",")
+    "{" + scalaSeqJsonized + "}"
+  }
+
+  private def scalaSeqToPgArray[T <: AnyRef: ClassTag](toConvert: Seq[T]): String = {
+    val scalaSeqJsonized = SerializationUtils.asJson(toConvert)  // this also correctly escapes double quotes
+    "{" + scalaSeqJsonized.substring(1, scalaSeqJsonized.length - 1) + "}"
```

Comment:
**Minor:**
```suggestion
    val scalaSeqJsonized = SerializationUtils.asJson(toConvert)                  // this also correctly escapes double quotes
    "{" + scalaSeqJsonized.substring(1, scalaSeqJsonized.length - 1) + "}"  //...only the square brackets have to be replaced with curly ones
```

Reply:
That's better, thanks, I'll change it :)